### PR TITLE
save and restore editor selection on model update

### PR DESCRIPTION
**Pull request by sheybey**
This fixes the cursor being moved to the beginning of the buffer on every edit.

This PR saves the selection before calling setText and restores the cursor index afterwards, clamping it at the end of the new buffer. However, this is still very slow for large config files since it is replacing the entire contents of the editor on every update. Maybe a better solution would be to let Quill manage its own state (model would be initialValue, onSave(newValue)?)

I am not familiar enough with vue/electron and their performance characteristics to make a more intelligent change here. This just fixes the cursor getting reset every keypress.

### CODE REVIEW

**Comment by sheybey** [comment]
#1008

File: src/components/QuillEditor.vue [hunk 1]
```diff
@@ -35,7 +35,11 @@
         @Watch("textValue")
         private updatedModel(newValue: string) {
             if (this.editor !== null) {
+                const selection = this.editor.getSelection();
                 this.editor.setText(newValue);
+                if (selection) {
+                    this.editor.setSelection(Math.min(newValue.length, selection.index), 0);
```
Comment (MythicManiac) [inline src/components/QuillEditor.vue:41]: I'm pretty sure you can omit the second parameter entirely as it defaults to 0

**Review by MythicManiac — APPROVED** [review]
I can't really comment on the performance aspect either since I don't know how the editor is implemented, but this sounds like an improvement. Thanks for the contribution!

**Comment by ebkr** [comment]
Thanks for the contribution! There was no mention of it being an issue before you pointed it out and it never used to behave like that, so I have no idea how long this been an issue for

**Pull request merged**